org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/AbstractImageBuilder.java

```diff
@@ -190,6 +190,15 @@ public void acceptResult(CompilationResult result) {
 				if (result.checkSecondaryTypes && !qualifiedTypeName.equals(compilationUnit.initialTypeName))
 					acceptSecondaryType(classFile);
 			}
+			for (int j = 0, l = this.javaBuilder.participants == null ? 0 : this.javaBuilder.participants.length; j < l; j++) {
+				BuildContext buildContext = new CompilationParticipantResult(compilationUnit,
+						this.compilationGroup == CompilationGroup.TEST);
+				byte[] postProcessingResult = this.javaBuilder.participants[j].postProcess(buildContext,
+						classFile.getBytes());
```

Comment:
Since it is safe to assume that the bytes should not be modified directly, but a new byte array is supposed to be made, I'd say the ByteArrayInputStream is a nice idea.